### PR TITLE
Add page explaining the variant products

### PR DESCRIPTION
**Pull request by NolwennP**
I added a new task to build pages which will only be accessible through error messages. They won't be any link to them in the doc website.
I also added the first page of documentation of this type: "What about product with variants?"

### CODE REVIEW

File: content/misc/products-with-variants.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+At this time, we already knew that we were going to make a big change in the way we manage the product with variants. So we knew that the notion of **variant groups might disappear**. That's why we decided to avoid developing the endpoints for this entity.
+
+But... Mistakes happen. And unfortunately, we forgot about the `variant_group` field that was inside the product format. :( 
+
+## The remodeling of the products with variants
+
+Handling the products with variants inside the PIM was, at that time, very **painful**. Our users could use the variant groups to gather similar products so as to enrich more easily their commun information.
+
+But they were **not user-friendly** because they could not appear directly in the product grid. You could not structure your variant groups thanks to families and there was no bulk actions or categorization of them.
```
Comment (LaurentPetard) [inline content/misc/products-with-variants.md]: there were no bulk actions

File: content/misc/products-with-variants.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  "family": "clothing",
+  "groups": [],
+  "variant_group": "my_structured_variant_group",
+  "categories": [ "tshirts"],
+  "enabled": true,
+  "values": {},
+  "created": "2017-10-05T11:25:48+02:00",
+  "updated": "2017-10-05T11:25:48+02:00",
+  "associations": {},
+  "metadata": {
```
Comment (skeleton) [inline content/misc/products-with-variants.md]: nope :)

File: content/misc/products-with-variants.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  "family": "clothing",
+  "groups": [],
+  "parent": "my_product_model",
+  "categories": ["tshirts"],
+  "enabled": true,
+  "values": {},
+  "created": "2017-10-05T11:25:48+02:00",
+  "updated": "2017-10-05T11:25:48+02:00",
+  "associations": {},
+  "metadata": {
```
Comment (skeleton) [inline content/misc/products-with-variants.md]: it's a EE data, are you sure you want to expose it here?

File: content/misc/products-with-variants.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+It depends on your products catalog structure for these products with variants.
+
+## My variant groups are very structured
+
+You used the variant groups is a very effective way and they are really structured:
+ - You have less than 5 axes in your variant group,
+ - The products inside your variant group use the same family,
+ - The products inside your variant group use the same attributes.
+
+It's very good news for you! The solution is to use the new product model and family variant entities, that were introduced in the remodeling of the products with variants for the 2.0.
```
Comment (skeleton) [inline content/misc/products-with-variants.md]: maybe you can add a link to the API reference for product model & family variant endpoints?

File: content/misc/products-with-variants.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+:::
+
+## My variant groups are a real mess
+
+Your variant groups are not structured enough:
+ - You have more than 5 axes in your variant group,
+ - The products inside your variant group does not use the same family,
+ - The products inside your variant group does not use the same attributes.
+
+You will only have one choice here: think about a better modeling for these products.
```
Comment (skeleton) [inline content/misc/products-with-variants.md]: :confident:

File: content/misc/products-with-variants.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+::: info
+For this solution to work, do not forget to create a product model with the code `my_product_model`. The family `clothing` used here, should have a family variants that is linked to this product model. ;)
+:::
+
+## My variant groups are a real mess
+
+Your variant groups are not structured enough:
+ - You have more than 5 axes in your variant group,
+ - The products inside your variant group does not use the same family,
```
Comment (skeleton) [inline content/misc/products-with-variants.md]: do not use

File: content/misc/products-with-variants.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+ - The products inside your variant group does not use the same attributes.
```
Comment (skeleton) [inline content/misc/products-with-variants.md]: same

File: content/misc/products-with-variants.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+}
+```
+
+As you can see, in the example above, there is no `variant_group` field anymore. So, how to migrate to 2.0, you will ask.
+
+It depends on your products catalog structure for these products with variants.
+
+## My variant groups are very structured
+
+You used the variant groups is a very effective way and they are really structured:
```
Comment (LaurentPetard) [inline content/misc/products-with-variants.md]: Misspell

File: content/misc/products-with-variants.md (hunk 5 above)
Comment (NolwennP) [reply content/misc/products-with-variants.md]: That's the rule Delphine gave me. We have to push our customers toward better catalog modeling. A catalog without a structure is a bad an inefficient catalog. :)

**Pull request merged**